corehq/apps/dump_reload/management/commands/print_domain_stats.py

```diff
@@ -0,0 +1,178 @@
+            sql = sql_counts.get(doc_type, '')
+            es = es_counts.get(doc_type, '')
+            output_rows.append((doc_type, couch, sql, es))
+
+        if csv:
+            self.output_csv(output_rows)
+        else:
+            self.output_table(output_rows)
+
+    def output_table(self, output_rows):
+        template = "{:<50} | {:<20} | {:<20} | {:<20}"
+        self._write_output(template, output_rows)
+
+    def output_csv(self, output_rows):
+        template = "{},{},{},{}\n"
+        self._write_output(template, output_rows, with_header_divider=False)
+
+    def _write_output(self, template, output_rows, with_header_divider=True, with_color=True):
+        self.stdout.write(template.format('Doc Type', 'Couch', 'SQL', 'ES'))
+        if with_header_divider:
+            self.stdout.write(template.format('-' * 50, *['-' * 20] * 3))
+        for doc_type, couch, sql, es in output_rows:
+            row_template = template
+            if with_color and es and ((couch and couch != es) or (sql and sql != es)):
+                row_template = shell_red(template)
+            self.stdout.write(row_template.format(doc_type, couch, sql, es))
+
+
+def _get_couchdb_counts(domain):
+    couch_db_counts = Counter()
+    for provider in DOC_PROVIDERS:
+        if isinstance(provider, DocTypeIDProvider):
+            for doc_type in provider.doc_types:
+                if doc_type == 'CommCareUser':
+                    continue  # want to split deleted
+                doc_class = get_document_class_by_doc_type(doc_type)
+                count = get_doc_count_in_domain_by_class(domain, doc_class)
+                couch_db_counts.update({doc_type: count})
+
+    for row in CommCareMultimedia.get_db().view('hqmedia/by_domain', key=domain, include_docs=False):
+        couch_db_counts.update(['CommCareMultimedia'])
+
+    mobile_user_count = get_mobile_user_count(domain)
+    couch_db_counts.update({
+        'WebUser': get_web_user_count(domain),
+        'CommCareUser': mobile_user_count,
+        'CommCareUser-Deleted': get_doc_count_in_domain_by_class(domain, CommCareUser) - mobile_user_count
+    })
+
+    # this is very slow, excluding for now
+    # for _, doc_ids in SyncLogIDProvider().get_doc_ids(domain):
+    #     couch_db_counts['SyncLog'] += len(doc_ids)
+    #
+    return couch_db_counts
+
+
+def _get_doc_counts_for_couch_db(couch_db, domain):
+    doc_types = couch_db.view(
+        "by_domain_doc_type_date/view",
+        startkey=[domain],
+        endkey=[domain, {}],
+        reduce=True,
+        group=True,
+        group_level=2
+    )
+
+    return Counter({row['key'][1]: row['value'] for row in doc_types})
+
+
+@allow_form_processing_queries()
+def _get_sql_counts(domain):
+    counter = Counter()
+    for model_class, queryset in get_querysets_to_dump(domain, []):
+        if model_class in (User, XFormInstanceSQL, CommCareCaseSQL):
+            continue  # User is very slow, others we want to break out
```

Comment:
case and form happen below (L126-127). I don't do Django users since we already have CommCareUser - I could add a command line option to include all which would include django users and SyncLogs.